package cgroupv1_release_agent_unknown_rootfs

import (
	"github.com/docker/docker/pkg/reexec"
	"github.com/pkg/errors"
	"github.com/ssst0n3/awesome_libs/awesome_error"
	"github.com/ssst0n3/awesome_libs/log"
	"os"
)

const ExpName = "cgroupv1-release_agent-unknown_rootfs"

func init() {
	reexec.Register(ExpName, Exploit)
}

func InvokeExploit(cmdOnHost string, debug bool) (err error) {
	debugArg := "--debug"
	if !debug {
		debugArg += "=false"
	}
	cmd := reexec.Command(ExpName, debugArg, "--cmd", cmdOnHost)
	log.Logger.Debug(cmd.Path)
	cmd.Stderr = os.Stderr
	cmd.Stdout = os.Stdout

	if err := cmd.Run(); err != nil {
		err = errors.Wrap(err, "cgroupv1_release_agent error on re-exec cmd")
		awesome_error.CheckErr(err)
		return err
	}
	return
}
